experimental/packages/exporter-logs-otlp-proto/test/node/OTLPLogExporter.test.ts

```diff
@@ -206,11 +204,10 @@ describe('OTLPLogExporter - node with proto over http', () => {
         done();
         return fakeRequest as any;
       });
+      collectorExporter.export(logs, () => {});
```

Comment:
This is a good question - I should've put a note here as it's quite odd :slightly_smiling_face: 

I removed the [`OTLPProtoExporterNodeBase`](https://github.com/open-telemetry/opentelemetry-js/blob/34387774caaa15307e8586206f1ca2e6df96605f/experimental/packages/otlp-proto-exporter-base/src/platform/node/OTLPProtoExporterNodeBase.ts) class and the proto exporters are now using the same export logic as the JSON exporter ([`OTLPExporterNodeBase`](https://github.com/open-telemetry/opentelemetry-js/blob/4f80adaa29d6d00a2d9238c7d67ebd46f2877a96/experimental/packages/otlp-exporter-base/src/platform/node/OTLPExporterNodeBase.ts)). That means is that it does not use the [`setImmediate()`](https://github.com/open-telemetry/opentelemetry-js/blob/34387774caaa15307e8586206f1ca2e6df96605f/experimental/packages/otlp-proto-exporter-base/src/platform/node/OTLPProtoExporterNodeBase.ts#L76) anymore to delay the export (see [here](https://github.com/open-telemetry/opentelemetry-js/blob/4f80adaa29d6d00a2d9238c7d67ebd46f2877a96/experimental/packages/otlp-exporter-base/src/platform/node/OTLPExporterNodeBase.ts#L86-L94) for code that's used now). As this `setImmedidate()` was there before, it was possible call export, then mock and wait for the test to finish. As it's gone, that does not work anymore (`export()` will actually export when it's called, not in the next tick).

From what I can tell the `setImmediate()` was added to `OTLPProtoExporterNodeBase` to avoid loading `http` before it can be instrumented (the comment mentioned `protobufjs`, but AFAIK there's no such instrumentation). While this was a good effort to try and delay the loading of the `http` module, `OTLPProtoExporterNodeBase` extended `OTLPExporterNodeBase` which made no effort to delay the loading of `http`. This made the proto-base's efforts to avoid loading ineffective, as it meant that `http` was imported when the base was required regardless of the `setImmediate()` in the `OTLPProtoExporterNodeBase`.

In follow-ups, I'll re-structure the transport code in the exporters to use the code from #4415. This will then *actually* implement the lazy-loading that was intended to be there in the first place.